cpp/src/arrow/compute/kernels/scalar_compare.cc

```diff
@@ -439,6 +472,224 @@ struct ScalarMinMax {
   }
 };
 
+template <typename Type, typename Op>
+Status ExecBinaryMinMaxScalar(KernelContext* ctx,
+                              const ElementWiseAggregateOptions& options,
+                              const ExecBatch& batch, Datum* out) {
+  if (batch.values.empty()) {
+    return Status::OK();
+  }
+  auto output = checked_cast<BaseBinaryScalar*>(out->scalar().get());
+  if (!options.skip_nulls) {
+    // any nulls in the input will produce a null output
+    for (const auto& value : batch.values) {
+      if (!value.scalar()->is_valid) {
+        output->is_valid = false;
+        return Status::OK();
+      }
+    }
+  }
+  const auto& first_scalar = *batch.values.front().scalar();
+  string_view result = UnboxScalar<Type>::Unbox(first_scalar);
```

Comment:
Hmm... if you use `checked_cast<const BaseBinaryScalar&>(first_scalar).view()`, you can probably remove the `Type` parametrization and produce less specializations of this function (which is always good for code size and compile times).